tikv-client/src/main/java/com/pingcap/tikv/region/RegionStoreClient.java

```diff
@@ -89,19 +93,49 @@ public int getValue() {
 
   private static final Logger logger = Logger.getLogger(RegionStoreClient.class);
   private TiRegion region;
+  private final RegionManager regionManager;
+  private final TiSession session;
+  public final LockResolverClient lockResolverClient;
   private TikvBlockingStub blockingStub;
   private TikvStub asyncStub;
-  private final RegionManager regionManager;
 
   public ByteString get(BackOffer backOffer, ByteString key, long version) {
-    Supplier<GetRequest> factory = () ->
-        GetRequest.newBuilder().setContext(region.getContext()).setKey(key).setVersion(version).build();
+    while (true) {
+      // we should refresh region
+      region = regionManager.getRegionByKey(key);
 
-    KVErrorHandler<GetResponse> handler =
-        new KVErrorHandler<>(
-            regionManager, this, region, resp -> resp.hasRegionError() ? resp.getRegionError() : null);
-    GetResponse resp = callWithRetry(backOffer, TikvGrpc.METHOD_KV_GET, factory, handler);
-    return getHelper(resp);
+      Supplier<GetRequest> factory = () ->
+          GetRequest.newBuilder().setContext(region.getContext()).setKey(key).setVersion(version).build();
+
+      KVErrorHandler<GetResponse> handler =
+          new KVErrorHandler<>(
+              regionManager, this, region, resp -> resp.hasRegionError() ? resp.getRegionError() : null);
+
+      GetResponse resp = callWithRetry(backOffer, TikvGrpc.METHOD_KV_GET, factory, handler);
+
+      if (resp.hasRegionError()) {
+        backOffer.doBackOff(BoRegionMiss, new RegionException(resp.getRegionError()));
+        continue;
+      }
+
+      if (resp.hasError()) {
+        if (resp.getError().hasLocked()) {
+          Lock lock = new Lock(resp.getError().getLocked());
+          boolean ok = lockResolverClient.resolveLocks(backOffer, new ArrayList<>(Arrays.asList(lock)));
+          if (!ok) {
+            // if not resolve all locks, we wait and retry
+            backOffer.doBackOff(BoTxnLockFast, new KeyException((resp.getError().getLocked().toString())));
+          }
+
+          continue;
+        } else {
+          // retryable or abort
```

Comment:
retriable？

Reply:
No, this means that the txn is retriable or aborted, we should instead rerunning this txn. I will add more comments.